theano/tensor/nnet/sigm.py

```diff
@@ -955,10 +960,13 @@ def local_inv_1_plus_exp(node):
             if len(nonconsts) == 1:
                 if nonconsts[0].owner and nonconsts[0].owner.op == tensor.exp:
                     if scalars and numpy.allclose(numpy.sum(scalars), 1):
-                        return opt._fill_chain(
+                        out = opt._fill_chain(
                             sigmoid(
                                 tensor.neg(nonconsts[0].owner.inputs[0])),
                             scalar_inputs)
+                        # keep stack trace
+                        copy_stack_trace(node.outputs[0], out)
```

Comment:
Here, we should combine the stack traces of `exp(x)`, `1 + exp(x)`, and `1/(1 + exp(x))`, respectively `nonconsts[0]`, `inv_arg` and `node.outputs[0]`. Can you add that with a comment?